### PR TITLE
using project(aresdb language CUDA) instead of using FindCUDA

### PR DESCRIPTION
**Pull request by lucafuji**
FindCUDA is a legacy way of compiling CUDA code, instead of we should using CUDA as a first class language in cmake

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,5 +1,23 @@
 cmake_minimum_required(VERSION 3.12)
-find_package(CUDA REQUIRED)
+# This has to be specified before enable_language according to
+# https://gitlab.kitware.com/cmake/cmake/issues/17559
```
Comment (voyager-dw) [inline CMakeLists.txt]: what is the context here? We use NVCC_FLAGS from CUDA package before, and we now remove that dependency?

Comment (lucafuji) [reply CMakeLists.txt]: the context is the title, for reference, refer to
https://cliutils.gitlab.io/modern-cmake/chapters/packages/CUDA.html

File: query/iterator_unittest.cu [hunk 2]
```diff
@@ -504,8 +503,7 @@ TEST(MeasureIteratorTest, CheckMax) {
 // cppcheck-suppress *
 TEST(RecordIDJoinIteratorTest, CheckIterator) {
   RecordID recordIDsH[5];
-  thrust::host_vector<ForeignTableIterator<int32_t>> vpItersHost(5);
-  ForeignTableIterator<int32_t> *vpIters = vpItersHost.data();
+  ForeignTableIterator<int32_t> vpItersH[5];
```
Comment (shz117) [inline query/iterator_unittest.cu]: are these test changes related to the cmake change?

Comment (lucafuji) [reply query/iterator_unittest.cu]: Some of the test changes are, like the bugs we have seen last time where nvcc hangs when compilation. 
Some of the test changes are just refactoring the code
